src/server/internalServer.h

```diff
@@ -62,21 +62,20 @@ struct GeoQuery {
 class SearchInfo {
   public:
     SearchInfo(const std::string& pattern);
-    SearchInfo(const std::string& pattern, GeoQuery geo_query);
+    SearchInfo(const std::string& pattern, GeoQuery geoQuery);
     SearchInfo(const RequestContext& request);
 
     zim::Query getZimQuery(bool verbose) const;
 
     friend bool operator<(const SearchInfo& l, const SearchInfo& r)
     {
-        return std::tie(l.bookName, l.pattern, l.has_geo_query, l.geo_query)
-             < std::tie(r.bookName, r.pattern, r.has_geo_query, r.geo_query); // keep the same order
+        return std::tie(l.bookName, l.pattern, l.geoQuery)
+             < std::tie(r.bookName, r.pattern, r.geoQuery); // keep the same order
     }
 
   public: //data
     std::string pattern;
-    bool has_geo_query;
-    GeoQuery geo_query;
+    std::pair<bool, GeoQuery> geoQuery;
```

Comment:
I don't like this. A better solution is to take advantage of the fact that `GeoQuery` can have an invalid state (for example when `distance < 0`) without adding a new data field to it. Add a proper default constructor to `GeoQuery` and a `bool isValid() const` member function.

Reply:
Done with a `operator bool` operator